### PR TITLE
match build failure conditions on staging & prod envs

### PR DESCRIPTION
**Pull request by mchelen-gov**
## Description
this PR does 3 things:
- fail build if broken links are found on staging build
- only send slack alert for broken links on `master` branch
- send list of broken links to slack channel

## Acceptance criteria
- [ ] things that break prod build should break staging build

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Comment by alexandec** [comment]
Looks good, thanks for addressing this! One other piece I noticed: https://github.com/department-of-veterans-affairs/vets-website/blob/master/jenkins/common.groovy#L170-L173
```
// Only break the build if broken links are found in master
if (IS_PROD_BRANCH || contentOnlyBuild) {
  throw new Exception('Broken links found')
}
```
Should this be updated as well?

**Review by alexandec — APPROVED** [review]

**Comment by mchelen-gov** [comment]
> Looks good, thanks for addressing this! One other piece I noticed: https://github.com/department-of-veterans-affairs/vets-website/blob/master/jenkins/common.groovy#L170-L173
> 
> ```
> // Only break the build if broken links are found in master
> if (IS_PROD_BRANCH || contentOnlyBuild) {
>   throw new Exception('Broken links found')
> }
> ```
> 
> Should this be updated as well?

@alexandec I think we still want it only on `master` branch, so that it's not checking PR branches. The goal is that builds which would fail on prod also fail on staging, so that nothing gets deployed to https://staging.va.gov/ which cannot be deployed to https://www.va.gov/

File: jenkins/common.groovy [hunk 1]
```diff
@@ -156,19 +156,21 @@ def checkForBrokenLinks(String buildLogPath, String envName, Boolean contentOnly
...
-    def linkCount = sh(returnStdout: true, script: "cd /application && wc -l ${csvFileName} | cut -d ' ' -f1") as Integer
-    slackSend message: "${linkCount} broken links found in the ${envName} build on `${env.BRANCH_NAME}`\n${env.RUN_DISPLAY_URL}".stripMargin(),
-      color: 'danger',
-      failOnError: true,
-      channel: 'cms-team'
+    echo "Found broken links."
 
     // Only break the build if broken links are found in master
     if (IS_PROD_BRANCH || contentOnlyBuild) {
+      echo "Notifying Slack channel."
```
Comment (ElijahLynn) [inline jenkins/common.groovy:163]: Nice, this seems like it should reduce some noise in the #cms-team channel!

**Review by indytechcook — APPROVED** [review]

**Review by swirtSJW — APPROVED** [review]
Some very helpful changes.  Thank you.

File: jenkins/common.groovy (hunk 1 above)
Comment (mchelen-gov) [reply jenkins/common.groovy:163]: yup, hopefully now the only broken link alerts will be ones we actually care about :D

**Pull request merged**